pkg/metrics/metrics.go

```diff
@@ -34,7 +34,11 @@ var log = logf.Log.WithName("metrics")
 // ExposeMetricsPort generate a Kubernetes Service to expose metrics port
 func ExposeMetricsPort() *v1.Service {
 	http.Handle("/"+k8sutil.PrometheusMetricsPortName, promhttp.Handler())
-	go http.ListenAndServe(":"+strconv.Itoa(k8sutil.PrometheusMetricsPort), nil)
+	go func() {
+		if err := http.ListenAndServe(":"+strconv.Itoa(k8sutil.PrometheusMetricsPort), nil); err != nil {
+			panic(err)
```

Comment:
Just my opinion: I think it's best to avoid `panic` as much as possible, because it's not helpful to the end user who will see it. But I see that `ExposeMetricsPort` uses it in several places. That being what it is, it's one thing to know that a particular call might result in a panic during startup, but after that you're ok. It's another thing for a ticking-panic-bomb to be sitting around all the time ready to explode at any moment.

We use a channel in the ansible operator to receive any errors from a long-running goroutine and then facilitate an orderly exit via `log.Fatal()`: https://github.com/operator-framework/operator-sdk/blob/afcc3721d3af72103b3e4180264a12cc61d56674/pkg/ansible/run.go#L82-L86

I think something like that would be a big improvement here. Ideally `ExposeMetricsPort` would return an error instead of the other `panic`s, and would perhaps take as an argument a `chan error` to pass into its goroutine.

Maybe that goes beyond the scope of what you're trying to do here, especially since this feature isn't currently being used. But I wanted to present this at least for thought.

Reply:
@mhrivnak Yeah I agree there's usually no good reason for using panic.
We're fixing this in #786 to return the error so we can log it at startup and then move on.

For this PR we probably don't need to change anything.

Reply:
Ok, I'll remove this change since it is being fixed in #786